src/rate_limiter.rs

```diff
@@ -248,7 +248,7 @@ impl RateLimiter {
             None => return None,
         };
 
-        let deduction = amount_charged.saturating_sub(part_count as u64 * MAX_PART_LEN as u64);
+        let deduction = amount_charged.saturating_sub(u64::from(part_count) * MAX_PART_LEN as u64);
```

Comment:
Shouldn't the `MAX_PART_LEN as u64` be also rewritten to `u64::from(MAX_PART_LEN)`?

Reply:
MAX_PART_LEN  is usize   and `u64::from(usize)` is not implemented  :(

Reply:
Got it. All good then.